### PR TITLE
[kubebuilder] Cluster deletion from orchestrator

### PR DESCRIPTION
**Pull request by AMecea**

### CODE REVIEW

File: pkg/controller/orchestrator/orchestrator_controller_test.go [hunk 1]
```diff
@@ -137,27 +150,41 @@ var _ = Describe("Orchestrator controller", func() {
...
 			c = mgr.GetClient()
-			recFn, requests = SetupTestReconcile(newReconciler(mgr))
+			recFn, requests = SetupTestReconcile(newReconciler(mgr, orcClient))
 			Expect(add(mgr, recFn)).To(Succeed())
 			stop = StartTestManager(mgr)
 
 			// wait a second for a request
 			Consistently(requests, reconcileTime).ShouldNot(Receive(Equal(expectedRequest)))
-			Eventually(requests).Should(Receive(Equal(expectedRequest)))
+			Eventually(requests, afterReconcileTime).Should(Receive(Equal(expectedRequest)))
```
Comment (calind) [inline pkg/controller/orchestrator/orchestrator_controller_test.go]: It think it's more obvious to write `2*reconcileTime` than `afterReconcileTime`

File: pkg/controller/orchestrator/orchestrator_controller_test.go [hunk 2]
```diff
@@ -137,27 +150,41 @@ var _ = Describe("Orchestrator controller", func() {
...
 			// delete the cluster
 			Expect(c.Delete(context.TODO(), cluster)).To(Succeed())
 
-			// wait a second for a request
-			// This is the request thats removes the finalizers from cluster
-			Consistently(requests, reconcileTime).ShouldNot(Receive(Equal(expectedRequest)))
-			Eventually(requests).Should(Receive(Equal(expectedRequest)))
+			// wait two seconds for a request
+			// This is the request thats unregister cluster from orchestrator
+			Eventually(requests, "1.2s").Should(Receive(Equal(expectedRequest)))
```
Comment (calind) [inline pkg/controller/orchestrator/orchestrator_controller_test.go]: Let's make this a multiple of `reconcileTime`. Also is there any reason for `1.2s`?

File: pkg/controller/orchestrator/orchestrator_controller_test.go [hunk 3]
```diff
@@ -137,27 +150,41 @@ var _ = Describe("Orchestrator controller", func() {
...
+			_, err := orcClient.Cluster(wCluster.GetClusterAlias())
+			Expect(err).ToNot(Succeed())
+
+			// this is the requests that removes the finalizer and then the
+			// cluster is deleted
+			Eventually(requests, "1.2s").Should(Receive(Equal(expectedRequest)))
 
 			// wait two seconds without request
-			Consistently(requests, 2*reconcileTime).ShouldNot(Receive(Equal(expectedRequest)))
+			Consistently(requests, "3s").ShouldNot(Receive(Equal(expectedRequest)))
```
Comment (calind) [inline pkg/controller/orchestrator/orchestrator_controller_test.go]: I think `3*reconcileTime` should suffice.

File: pkg/controller/orchestrator/orchestrator_controller_test.go [hunk 4]
```diff
@@ -137,27 +150,41 @@ var _ = Describe("Orchestrator controller", func() {
...
+			// This is the request thats unregister cluster from orchestrator
+			Eventually(requests, "1.2s").Should(Receive(Equal(expectedRequest)))
+
+			wCluster := wrapcluster.NewMysqlClusterWrapper(cluster)
+			_, err := orcClient.Cluster(wCluster.GetClusterAlias())
+			Expect(err).ToNot(Succeed())
+
+			// this is the requests that removes the finalizer and then the
+			// cluster is deleted
+			Eventually(requests, "1.2s").Should(Receive(Equal(expectedRequest)))
```
Comment (calind) [inline pkg/controller/orchestrator/orchestrator_controller_test.go]: Can we also make this a multiple of reconcileTime?

File: pkg/controller/orchestrator/orchestrator_controller_test.go (hunk 1 above)
Comment (AMecea) [reply pkg/controller/orchestrator/orchestrator_controller_test.go]: `afterReconcileTime` is `200 * time.Milisecond`

**Pull request merged**